scripts/api/user.ts

```diff
@@ -18,8 +19,13 @@ function isLoggedIn() {
     return session?.identity?._id != null;
 }
 
+function getAll(): Array<IUser> {
+    return ng.get('desks').users._items;
+}
+
 export const user = {
     hasPrivilege,
     isLoggedIn,
     getCurrentUserId,
+    getAll,
```

Comment:
Can you export an `OrderedMap<IUser['_id'], IUser]>` from the API? So we also have a capability to get a user by ID without having to iterate.

Reply:
It works quite well with vocabularies which I did a while ago https://github.com/superdesk/superdesk-client-core/blob/develop/scripts/api/vocabularies.ts#L97

When thinking more about it, I'm not sure whether it's better to expose an ordered map or simply have 2 methods - getAll that would return an array and getById that would return a single item - without involving immutable.

Reply:
I think it makes sense to get the map, imo when you get all users it's usually rather for a lookup and not for listing them, this is rather an edge case.

Reply:
Functionally it works great both - lookup and iteration. The only thing I'm wary about is developers being confused that getting all items when they only need one would be expensive in terms of performance when in reality it wouldn't.